### PR TITLE
test: Add testToIso8601StringNil_ReturnsNil

### PR DESCRIPTION
**Pull request by philipphofmann**
Add a test to ensure we don't crash when passing nil to sentry_toIso8601String.

Came up while investigating https://github.com/getsentry/sentry-cocoa/issues/4582.

#skip-changelog

### CODE REVIEW

File: Tests/SentryTests/SentryTests.m [hunk 1]
```diff
@@ -130,4 +130,13 @@ - (void)testDateCategoryCompactibility
     XCTAssertEqual([date timeIntervalSince1970], 1582803326.0);
 }
 
+- (void)testToIs08691StringNil_ReturnsNil
```
Comment (philprime) [inline Tests/SentryTests/SentryTests.m:133]:
```suggestion
- (void)testToIso8691StringNil_ReturnsNil
```

**Review by philprime — APPROVED** [review]
Thanks for adding this test, but why is this not part of #4982?

**Comment by philipphofmann** [comment]
> Thanks for adding this test, but why is this not part of #4982?

This is toISO the PR above is about fromISO. Two different methods.
